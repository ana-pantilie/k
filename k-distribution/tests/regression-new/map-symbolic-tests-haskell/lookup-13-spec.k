// Copyright (c) 2019 K Team. All Rights Reserved.
// generated by gen-tests.sh

requires "map-tests.k"

module LOOKUP-13-SPEC
    imports MAP-TESTS

    claim <k> lookup ( (MAP:Map X:MyId |-> 1) [Y:MyId <- 2] [ Z:MyId ] ) => lookupResult ( MAP ( X |-> 1 ) [ Z ] ) </k> requires Z =/=K Y

endmodule
