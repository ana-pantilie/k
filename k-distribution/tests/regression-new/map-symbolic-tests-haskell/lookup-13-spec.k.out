  #Not ( {
    ( MAP
    X:MyId |-> 1 ) [ Y:MyId <- 2 ] [ Z:MyId ]
  #Equals
    MAP
    ( X:MyId |-> 1 ) [ Z:MyId ]
  } )
#And
  #Not ( {
    Y
  #Equals
    Z
  } )
#And
  <k>
    lookupResult ( ( MAP
    X:MyId |-> 1 ) [ Y:MyId <- 2 ] [ Z:MyId ] ~> . ) ~> .
  </k>
#And
  {
    false
  #Equals
    X:MyId in_keys ( MAP )
  }
#And
  {
    true
  #Equals
    Z:MyId in_keys ( MAP
    ( X:MyId |-> 1 ) )
  }
